src/main/tut/2.5-the-yoneda-lemma.md

```diff
@@ -0,0 +1,122 @@
+```Haskell
+αx :: C(a, x) -> F x
+αy :: C(a, y) -> F y
+```
+```scala
+def αx[A, X]: C[A, X] => F[X]
+def αy[A, Y]: C[A, Y] => F[Y]
+```
+................
+```Haskell
+Nat(C(a, -), F) ≅ F a
+```
+```scala
+C[A, ?] ~> F ≅ F[A]
+```
+................
+```Haskell
+C(a, g) :: C(a, a) -> C(a, x)
+```
+```scala
+def cg[A, X]: C[A, A] => C[A, X]
+```
+................
+```Haskell
+F g :: F a -> F x
+```
+```scala
+def fg[A, X]: F[A] => F[X]
+```
+................
+```Haskell
+type Reader a x = a -> x
+```
+```tut:silent
+type Reader[A, X] = A => X
+```
+................
+```Haskell
+instance Functor (Reader a) where
+    fmap f h = f . h
+```
+```scala
+implicit def readerFunctor[A] =
+  new Functor[({type T[X] = Reader[A, X]})#T] {
+    def fmap[X, B](f: X => B)(h: Reader[A, X])
+        : Reader[A, B] =
+      f compose h
+  }
+
+/* type lambda with Kind Projector plugin:
+   new Functor[Reader[A, ?]] {...} */
+```
+................
+```Haskell
+alpha :: forall x . (a -> x) -> F x
+```
+```scala
+/*
+In order to make a universal transformation,
+another type needs to be introduced.
+To read more about FunctionK (~>):
+typelevel.org/cats/datatypes/functionk.html
+*/
+trait ~>[F[_], G[_]] {
+  def apply[B](fa: F[B]): G[B]
+}
+
+/*
+Kind Projector plugin provides
+a more concise syntax for type lambdas:
+*/
+def alpha[A]: (A => ?) ~> F
+```
+................
+```Haskell
+forall x . (a -> x) -> F x ≅ F a
+```
+```scala
+(A => ?) ~> F ≅ F[A]
```

Comment:
should it be something like `def α[A]: (A => ?) ~> F ≅ F[A]` ?

Reply:
that would translate to 
```Haskell
forall x a . (a -> x) -> F x ≅ F a
```
Right?